dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -111,43 +122,20 @@ internal class DatadogRumMonitor(
             eventData,
             attributes
         )
-        activeResources[key] = event
+        activeResources[WeakReference(key)] = event
```

Comment:
How does it work in Kotlin? Is this record removed from this mutable map after `key` is deallocated and GC triggers? Seems weird to me, as after GC there will be still a valid memory allocated for `WeakReference` object that will keep this entry alive, but I dunno how Kotlin optimises it internally. Shouldn't `WeakHashMap` be used instead?

Reply:
WeakReference is a Java construct that holds a weak reference to the object, which means it is not "counted" when the GC looks at the key. That way it can clean it up, and the WeakReference will then point to null.